packages/aws-cdk-lib/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -1840,6 +1880,38 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
 
     return errors;
   }
+
+  private renderInstanceMaintenancePolicy(
+    maxHealthPercentage?: number,
+    minHealthPercentage?: number,
+  ): CfnAutoScalingGroup.InstanceMaintenancePolicyProperty | undefined {
+    if (maxHealthPercentage === undefined && minHealthPercentage === undefined) return;
+    if (maxHealthPercentage === undefined || minHealthPercentage === undefined) {
+      throw new Error(`Both or neither of maintenancePolicyMinHealthPercentage and maintenancePolicyMaxHealthPercentage must be specified, got maintenancePolicyMinHealthPercentage: ${minHealthPercentage} and maintenancePolicyMaxHealthPercentage: ${maxHealthPercentage}`);
+    }
+    if (maxHealthPercentage === -1 && minHealthPercentage === -1) {
+      return {
+        maxHealthyPercentage: maxHealthPercentage,
+        minHealthyPercentage: minHealthPercentage,
+      };
+    }
+    if (minHealthPercentage === -1 || maxHealthPercentage === -1) {
+      throw new Error(`Both maintenancePolicyMinHealthPercentage and maintenancePolicyMaxHealthPercentage must be -1 to clear the previously set value, got maintenancePolicyMinHealthPercentage: ${minHealthPercentage} and maintenancePolicyMaxHealthPercentage: ${maxHealthPercentage}`);
+    }
```

Comment:
```suggestion
    if (minHealthPercentage === -1 || maxHealthPercentage === -1 && minHealthPercentage !== maxHealthPercentage) {
      throw new Error(`Both maintenancePolicyMinHealthPercentage and maintenancePolicyMaxHealthPercentage must be -1 to clear the previously set value, got maintenancePolicyMinHealthPercentage: ${minHealthPercentage} and maintenancePolicyMaxHealthPercentage: ${maxHealthPercentage}`);
    }
```

This would be equivalent and remove an additional if condition, but isn't a huge deal.

Reply:
No, in the case for the code and  both maxHealthPercentage and minHealthPercentage is `-1`, the next branch causes an error.

```ts
    if (maxHealthPercentage < 100 || maxHealthPercentage > 200) {
      throw new Error(`maxHealthPercentage must be between 100 and 200, or -1 to clear the previously set value, got ${maxHealthPercentage}`);
    }
    if (minHealthPercentage < 0 || minHealthPercentage > 100) {
      throw new Error(`minHealthPercentage must be between 0 and 100, or -1 to clear the previously set value, got ${minHealthPercentage}`);
    }
```

So, if both are `-1`, we need to return them as soon as possible.

Reply:
Or, the following code would work. I will go with this!

```ts
    if ((minHealthPercentage === -1 || maxHealthPercentage === -1) && minHealthPercentage !== maxHealthPercentage) {
      throw new Error(`Both minHealthPercentage and maxHealthPercentage must be -1 to clear the previously set value, got minHealthPercentage: ${minHealthPercentage} and maxHealthPercentage: ${maxHealthPercentage}`);
    }
    if (maxHealthPercentage !== -1 && (maxHealthPercentage < 100 || maxHealthPercentage > 200)) {
      throw new Error(`maxHealthPercentage must be between 100 and 200, or -1 to clear the previously set value, got ${maxHealthPercentage}`);
    }
    if (minHealthPercentage !== -1 && (minHealthPercentage < 0 || minHealthPercentage > 100)) {
      throw new Error(`minHealthPercentage must be between 0 and 100, or -1 to clear the previously set value, got ${minHealthPercentage}`);
    }
```

Reply:
I changed.